TACC/tup-ui
# fix: reduce length of post_nomials data

## Overview & Changes

1. Reduce `post_nomials` to 50 characters.
2. Manually squash all migrations into one.

## Related

- polish #253

## Testing

0. Have CMS running on this branch.
1. Migrate everything for this plugin.
	1. `git checkout 2b3f0a1`
	2. `docker exec -it tup_cms /bin/bash`
    	```
    	python manage.py migrate staff_profile
    	```
2. Update database to use new squashed migration.
	1. `git checkout a3819d6`
	2. `docker exec -it tup_cms /bin/bash`
    	```
    	python manage.py migrate --fake staff_profile 0001_squashed_for_release
    	```
3. Restart CMS server.
4. Open http://localhost:8000/.
5. Review any existing Staff Profile plugin instances you have.
6. Confirm the data is all still present.
	<sup>Depending on your CMS history, the images may be gone. I do not know if this is related to migrations. If you edit the image (or go in Filer to the folder where it should be), all your image files may be present but data to render them as images is unavailable.[^1]</sup>

## UI

All my various staff profile plugin instances retained their data. (But images are broken.)[^1]

| a | b | c | d |
| - | - | - | - |
| ![Screenshot 2023-08-24 at 3 41 24 PM](https://github.com/TACC/tup-ui/assets/62723358/b2e15222-fe6c-4a5a-8e7f-6ffa842b772f) | ![Screenshot 2023-08-24 at 3 41 30 PM](https://github.com/TACC/tup-ui/assets/62723358/97ebfff5-f6bf-471c-aecc-c856420c50c8) | ![Screenshot 2023-08-24 at 3 41 40 PM](https://github.com/TACC/tup-ui/assets/62723358/c600045b-5ed4-4f60-bab5-0532b2aa3430) | ![Screenshot 2023-08-24 at 3 41 47 PM](https://github.com/TACC/tup-ui/assets/62723358/4ffaee4a-523a-4654-882b-97a43812014a) |

[^1]: Maybe the broken images are form me taking down my containers and starting new ones for other CMS projects.
